destination/writer.go

```diff
@@ -0,0 +1,164 @@
+// Copyright © 2023 Meroxa, Inc. & Yalantis
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package destination
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/data/azcosmos"
+	"github.com/conduitio-labs/conduit-connector-cosmos-nosql/common"
+	sdk "github.com/conduitio/conduit-connector-sdk"
+)
+
+// Writer implements a writer logic for the Neo4j Destination.
+type writer struct {
+	containerClient *azcosmos.ContainerClient
+	partitionKey    azcosmos.PartitionKey
+}
+
+// New creates a new instance of the [Writer].
+func newWriter(config Config) (*writer, error) {
+	w := &writer{
+		partitionKey: azcosmos.NewPartitionKeyString(config.PartitionValue),
+	}
+
+	// create an KeyCredential containing the account's primary key
+	cred, err := azcosmos.NewKeyCredential(config.PrimaryKey)
+	if err != nil {
+		return nil, fmt.Errorf("new key credential: %w", err)
+	}
+
+	// create a new instance of Cosmos client
+	client, err := azcosmos.NewClientWithKey(config.URI, cred, nil)
+	if err != nil {
+		return nil, fmt.Errorf("new client: %w", err)
+	}
+
+	// create a new instance of the specific container client
+	w.containerClient, err = client.NewContainer(config.Database, config.Container)
+	if err != nil {
+		return nil, fmt.Errorf("new container client: %w", err)
+	}
+
+	return w, nil
+}
+
+// Write writes a record to the destination.
+func (w *writer) Write(ctx context.Context, record sdk.Record) error {
+	err := sdk.Util.Destination.Route(ctx, record,
+		w.create,
+		w.replace,
+		w.delete,
+		w.create,
+	)
+	if err != nil {
+		return fmt.Errorf("route record: %w", err)
+	}
+
+	return nil
+}
+
+func (w *writer) create(ctx context.Context, record sdk.Record) error {
+	payload := make(sdk.StructuredData)
+	if err := json.Unmarshal(record.Payload.After.Bytes(), &payload); err != nil {
+		return fmt.Errorf("unmarshal payload: %w", err)
+	}
+
+	// if sdk.Record.Key does not contain the 'id' key
+	if _, okPayload := payload[common.KeyID]; !okPayload {
```

Comment:
we have comment `if sdk.Record.Key does not contain the 'id' key` but we check id in payload and not in key. Is it ok?

Reply:
Yeah, you're right 👍 
Fixed!